src/Infrastructure.VS/ThreadHandling.cs

```diff
@@ -48,6 +48,10 @@ public class ThreadHandling : IThreadHandling
 
         public async Task RunOnUIThread(Action op) => await VS.RunOnUIThread.RunAsync(op);
 
+        // TODO: check the implementation of RunOnUIThread.Run is actually synchronous
+        // See https://github.com/SonarSource/sonarlint-visualstudio/issues/4179
+        public void RunOnUIThreadSync(Action op) => VS.RunOnUIThread.Run(op);
```

Comment:
This is a bit of a worry. However, fixing the method will change the runtime behaviour of all of the exiting callers and will need to be looked at carefully, so I created a separate ticket.